### PR TITLE
Feat platform client 1865 clean up todo s

### PR DESCRIPTION
**Pull request by KZavorotna**
Is required for https://github.com/withthegrid/platform-client/pull/1949

### CODE REVIEW

File: src/routes/graph/add-pin.ts [hunk 1]
```diff
@@ -40,10 +44,12 @@ const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
     fields: fieldsToServerFullSchema.required().example({ id: 'My port' }),
     typeKey: Joi.string().allow(null).default(null),
     edgeHashId: Joi.string().allow(null).default(null).example('ka08d'),
+    pinGridsHashIds: Joi.array().items(Joi.string()).allow(null).default(null),
   }).required(),
   right: { environment: 'STATIC' },
   response: Joi.object().keys({
     pin: pinSchema.required(),
+    grids: Joi.array().items(gridSchema),
```
Comment (Arinono) [inline src/routes/graph/add-pin.ts:52]: **requested change**

To match the TS signature:
```suggestion
    grids: Joi.array().items(gridSchema).required(),
```

File: src/routes/graph/get-pin-group.ts [hunk 2]
```diff
@@ -44,7 +47,10 @@ const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
   right: { environment: 'READ' },
   response: (apiVersion: number): Joi.ObjectSchema => Joi.object().keys({
     pinGroup: pinGroupSchema.required(),
-    pins: Joi.array().items(pinSchema).required(),
+    pins: Joi.array().items(Joi.object().keys({
+      pin: pinSchema,
+      grids: Joi.array().items(gridSchema),
```
Comment (Arinono) [inline src/routes/graph/get-pin-group.ts:52]: **requested change**

to match the Ts signature:
```suggestion
      grids: Joi.array().items(gridSchema).required(),
```

File: src/routes/graph/update-pin.ts [hunk 3]
```diff
@@ -26,9 +29,11 @@ const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
   body: Joi.object().keys({
     fields: fieldsToServerUpdateSchema.example({ id: 'My port' }),
     edgeHashId: Joi.string().allow(null).example('ka08d'),
+    pinGridsHashIds: Joi.array().items(Joi.string()),
   }).required(),
   response: Joi.object().keys({
     pin: pinSchema.required(),
+    grids: Joi.array().items(gridSchema),
```
Comment (Arinono) [inline src/routes/graph/update-pin.ts:36]: **requested change**

to match the TS signature:
```suggestion
    grids: Joi.array().items(gridSchema).required(),
```

**Review by Arinono — CHANGES_REQUESTED** [review]

**Review by Arinono — APPROVED** [review]
as far as I can tell, it looks good to me. I'll let Gabi have the final word depending on his feedback on the platform

File: src/routes/graph/add-pin.ts [hunk 4]
```diff
@@ -23,11 +25,13 @@ interface EffectiveRequest {
     fields: FieldsToServerFull;
     typeKey: string | null;
     edgeHashId: string | null;
+    pinGridsHashIds: string[];
```
Comment (srieding) [inline src/routes/graph/add-pin.ts:28]: Should this not read | null if you add null as the default value for pinGridHashIds?

Comment (Arinono) [reply src/routes/graph/add-pin.ts:28]: Yes, I think it should too, at least to match the Joi definition. I'll open a PR

Comment (Arinono) [reply src/routes/graph/add-pin.ts:28]: https://github.com/withthegrid/platform-sdk/pull/460

**Pull request merged**